packages/core/src/tools/mcp-client.ts

```diff
@@ -62,6 +71,11 @@ export enum MCPDiscoveryState {
  */
 const mcpServerStatusesInternal: Map<string, MCPServerStatus> = new Map();
 
+/**
+ * Map to track discovered prompts for each MCP server
+ */
+const mcpServerPromptsInternal: Map<string, DiscoveredMCPPrompt[]> = new Map();
```

Comment:
I feel like the statusesInternal format above is being a bit too verbose as well. I see you followed that though, ideally these would both be `serverStatuses` and `prompts`